### PR TITLE
Encode gif if CameraMode is set to GIF

### PR DESCRIPTION
**Pull request by Deco354**
## What it does

GIFs sometimes aren't encoded when the CameraMode is set to gif. This is as there's a `settings.feature.gifs` flag we check that determines if gifs should be enabled.

It's possible to use the API to trigger a CameraViewController on gif mode without having this feature on. This is [a bug](https://github.com/tumblr/kanvas-ios/issues/160) in itself but for the time being this PR ensures that if a user has selected the gif camera and taken a photo we always encode that gif. Prior to now the gif was coming out as a video.

I've made this function testable and added a test. The commit with the actual functionality change is below:
[The fix commit](https://github.com/tumblr/kanvas-ios/pull/159/commits/a66a7e4179d588fcd6deef88f3dcf3461eefa158)

## How to test

Use a real device for both tests

### Example project

* Load the example project on to a real device
* Disable gif support
* Change the camera to gif mode
* Take a photo and post
* You should be brought back to the dashboard
* Repeat this 4-5 times to check we don't crash or freeze

### Tumblr Integration
 
[Test instructions + JIRA Ticket](https://github.tumblr.net/TumblrMobile/orangina/pull/25164)

## Etc

CC: @bjtitus I'd like your eyes on [the fix commit](https://github.com/tumblr/kanvas-ios/pull/159/commits/a66a7e4179d588fcd6deef88f3dcf3461eefa158). I'm making what I think to be a safe assumption that we'd always want to encode a gif once the user has set the camera to gif mode. If this isn't the case I may need to expand this PR to [prevent us getting to this state in the first place](https://github.com/tumblr/kanvas-ios/issues/160)

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

**Review by mindgraffiti — APPROVED** [review]
- 1 nit
- tested on device using example app (do you want this tested in Orangina too? I didn't do that part)

I tried out several of the filters and "plain" gifs and didn't encounter any weirdness or bugs.

**Review by mindgraffiti — APPROVED** [review]
- 1 nit
- tested on device using example app (do you want this tested in Orangina too? I didn't do that part)

I tried out several of the filters and "plain" gifs and didn't encounter any weirdness or bugs.

File: Example/KanvasExampleTests/CameraPreview/CameraPreviewControllerTests.swift [hunk 1]
```diff
@@ -66,9 +64,20 @@ final class CameraPreviewControllerTests: FBSnapshotTestCase {
...
+                           assetsHandler: AssetsHandlerType? = nil,
+                           cameraMode: CameraMode? = nil,
+                           gifEncoder: GIFEncoder = GIFEncoderImageIO()) -> CameraPreviewViewController {
+        
         let handler = assetsHandler ?? AssetsHandlerStub()
-        let viewController = CameraPreviewViewController(settings: settings, segments: segments, assetsHandler: handler, cameraMode: cameraMode)
+        let viewController = CameraPreviewViewController(settings: settings,
+                                                         segments: segments,
+                                                         assetsHandler:
+                                                            handler,
```
[the remark this answers is not in the document]
Comment (Deco354) [reply Example/KanvasExampleTests/CameraPreview/CameraPreviewControllerTests.swift:78]: Nice catch. 👍

**Review by bjtitus — APPROVED** [review]
It's been a while since I looked at any video export stuff but I think this makes sense. Thanks for making the change!

✅ I was able to encode GIFs without enabling the GIF setting.

**Comment by Deco354** [comment]
> do you want this tested in Orangina too? I didn't do that part

Yes please if you get the chance. The PR is primarily to fix an issue on Orangina so it's probably best we test it now rather than in the Orangina PR which may require another Kanvas PR if the testing doesn't work.

Unless @bjtitus did the Orangina testing in which case I think we're good to go.

**Comment by bjtitus** [comment]
@Deco354 I just now verified the Orangina PR using that branch and the test instructions over there. Everything looks good ✅

**Comment by Deco354** [comment]
> I have considered if this change warrants release notes and have added them to the appropriate section in the CHANGELOG.md if necessary.

I'll do this within the version bump PR

**Pull request merged**